packages/snaps-controllers/src/snaps/SnapController.test.tsx

```diff
@@ -8801,6 +8858,7 @@ describe('SnapController', () => {
       );
 
       const newState = { myVariable: 2 };
+      const promise = waitForStateChange(messenger);
```

Comment:
Why do we need to wait for state changes here?

Reply:
We do assertions based on `snapController.state.snapStates`, which is only updated when the state is persisted. Waiting for a state change is an easy way to ensure the state is persisted, and avoids `sleep` with arbitrary times.